system/classes/KO7/REST/Format/XML.php

```diff
@@ -0,0 +1,41 @@
+<?php
+/**
+ * Class for formatting REST-Response bodies as XML
+ *
+ * @package KO7\REST
+ *
+ * @copyright  (c) since 2016 Koseven Team
+ * @license    https://koseven.ga/LICENSE
+ */
+class KO7_REST_Format_XML extends REST_Format {
+
+    /**
+     * Format function
+     *
+     * @param array $body Body to format
+     *
+     * @throws REST_Exception
+     *
+     * @return string
+     */
+    public function format(array $body): string
+    {
+        // Check if php-xml is loaded
+        if ( ! extension_loaded('xml'))
+        {
+            throw new REST_Exception('PHP XML Module not loaded.');
+        }
+
+        // Create new XML Element
+        $xml = new SimpleXMLElement('<root/>');
+
+        // Add Child foreach body element
+        $xml = $xml->addChild('data');
+        array_walk_recursive($data, static function($value, $key) use ($xml, &$result) {
```

Comment:
1. $result` not used
2. wrong xml, add tests

Reply:
~~~
$data = ['a' => ['a2' => ['a3' => 1]], 'b' => 1];
~~~
$xml->asXml():
~~~
<data><a3>1</a3><b>1</b></data>
~~~

Reply:
Agrre, thanks, will fix this, i write tests when code is done, therefore i do not waste time rewriting them everytime

Reply:
~~~
/**
 * Convert array to xml.
 * 
 * @param array $data
 * @param string|SimpleXMLElement $root
 * @param string $charset
 * @return SimpleXMLElement
 */
public function arrayToXml(array $data, $root, $charset = null)
{ 
    if (is_string($root)) {
        $root = new SimpleXMLElement(
            '<?xml version="1.0" encoding="'.($charset ?: KO7::$charset).'"?><'.$root.'/>',
            LIBXML_BIGLINES | LIBXML_NOCDATA | LIBXML_NOENT
        );
    }
    
    foreach ($data as $name => $value) {
        if (substr($name, 0, 1) == '@') {
            $root->addAttribute($name, $value);
        } else {
            if (is_array($value)) {
                $this->arrayToXml($value, $root->addChild($name), $charset);
            } else {
                $root->addChild($name, $value);
            }
        }
    }
    
    return $root;
}
~~~

Reply:
Fixed, also fixed the content-type